resources/generate_book.py

```diff
@@ -659,6 +659,62 @@ def make_book(rm: ResourceManager, i18n: I18n, local_instance: bool = False):
+            crafting('tfc:crafting/wood/birch_scribing_table', text_contents='The scribing table crafting recipe.'),
+            image('tfc:textures/gui/book/gui/scribing.png', text_contents='The scribing screen takes text entry at the top, an input on the left, a dye in the center. The output is taken from the right slot.'),
+            block_spotlight('', 'The scribing table.', 'tfc:wood/scribing_table/kapok')
+        )),
+        entry('advanced_building_materials', 'Advanced Materials', 'tfc:brick/rhyolite', pages=(
+            text('As you progress in TFC, you may want for stronger and prettier building materials, as well as some devices you may remember from vanilla. One such material is $(thing)Stone Bricks$(), which are made with $(thing)Bricks$() and $(thing)Mortar$()'),
+            instant_barrel_recipe('tfc:barrel/limewater', 'To make mortar, $(thing)Limewater$() must first be made with $(thing)Water$() and $(l:mechanics/flux)Flux$() in a $(l:mechanics/barrel)Barrel$().'),
+            sealed_barrel_recipe('tfc:barrel/mortar', 'Then, combine it with $(thing)Sand$() to make mortar.'),
+            crafting('tfc:crafting/rock/gneiss_brick', text_contents='Bricks can be crafted from loose rocks in bulk with a $(l:mechanics/chisel)Chisel$(). These rocks can be made into $(thing)Pressure Plates$() and $(thing)Buttons$()'),
+            crafting('tfc:crafting/rock/gneiss_pressure_plate', 'tfc:crafting/rock/gneiss_button'),
+            crafting('tfc:crafting/rock/gneiss_bricks', text_contents='Finally, bricks and mortar can be combined into bricks. Bricks can be made into slabs, stairs, and walls.'),
+            crafting('tfc:crafting/rock/gneiss_cracked', text_contents='Bricks can turned into cracked bricks with a $(thing)Hammer$().'),
+            crafting('tfc:crafting/rock/quartzite_hardened', text_contents='Mortar can be used to make $(thing)Hardened Stone$(), a kind of raw stone that does not collapse.'),
+        )),
+        entry('salad', 'Salads', 'tfc:food/protein_salad', pages=(
+            text('$(thing)Salads$() are a meal prepared in a $(thing)Bowl$() from up to five $(thing)Fruits$(), $(thing)Vegetables$(), or $(thing)Cooked Meats$(). Salads are made in a special salad screen, created with a $(thing)Bowl$().'),
+            clay_knapping('tfc:clay_knapping/bowl_4', 'Ceramic bowls are made through the knapping and firing of clay.'),
+            image('tfc:textures/gui/book/gui/salad.png', text_contents='The salad screen is opened by pressing $(item)$(k:key.use)$() while holding $(item)$(k:key.sneak)$().'),
+            text('When you are ready, take your salad out of the slot on the right side.$(br)$(br)The $(l:the_world/food_and_water)Nutrients$(), Water, and Saturation of a salad are 75% of the total of all nutrients of its ingredients. When a salad is made, its decay refreshes.')
+        )),
```

Comment:
Re-order these pages? Start text() with `The salad screen is...`, and then depending on length the caption on the image can be "The Salad Interface" (consistent with other gui spotlights)